be/src/olap/task/engine_publish_version_task.cpp

```diff
@@ -0,0 +1,133 @@
+
+        Version version(partitionVersionInfo.version, partitionVersionInfo.version);
+        VersionHash version_hash = partitionVersionInfo.version_hash;
+
+        // each tablet
+        for (auto& tablet_rs : tablet_related_rs) {
+            OLAPStatus publish_status = OLAP_SUCCESS;
+            TabletInfo tablet_info = tablet_rs.first;
+            RowsetSharedPtr rowset = tablet_rs.second;
+            LOG(INFO) << "begin to publish version on tablet. "
+                    << "tablet_id=" << tablet_info.tablet_id
+                    << ", schema_hash=" << tablet_info.schema_hash
+                    << ", version=" << version.first
+                    << ", version_hash=" << version_hash
+                    << ", transaction_id=" << transaction_id;
+            TabletSharedPtr tablet = TabletManager::instance()->get_tablet(tablet_info.tablet_id, tablet_info.schema_hash);
+
+            if (tablet.get() == NULL) {
+                LOG(WARNING) << "can't get tablet when publish version. tablet_id=" << tablet_info.tablet_id
+                             << "schema_hash=" << tablet_info.schema_hash;
+                _error_tablet_ids->push_back(tablet_info.tablet_id);
+                res = OLAP_ERR_PUSH_TABLE_NOT_EXIST;
+                continue;
+            }
+
+            // publish version, one tablet only has one rowset for a single txn
+            // if one tablet has more than one rowset for a single txn, 
+            // should consider part of rowset meta persist 
+            // failed, it is very difficault, so that not deal with this case
+            // get rowsets from txn manager according to tablet and txn id
+            // set rowset version 
+            // persist rowset meta
+            rowset->set_version(version);
+            rowset->set_version_hash(version_hash);
+            // TODO(ygl): currently, tablet meta will be persist when add a new rowset,
+            // so that not persist rowset meta here
+            /**
+            publish_status = RowsetMetaManager::save(
+                tablet->data_dir()->get_meta(),
+                rowset->rowset_id(),
+                rowset->rowset_meta());
+            if (publish_status != OLAP_SUCCESS) {
+                LOG(WARNING) << "save pending rowset failed. rowset_id:" << rowset->rowset_id() 
+                             << "tablet id: " << tablet_info.tablet_id
+                             << "txn id:" << transaction_id;
+                error_tablet_ids->push_back(tablet_info.tablet_id);
+                res = OLAP_ERR_ROWSET_SAVE_FAILED;
+                continue;
+            }
+            */
+            // add visible rowset to tablet
+            publish_status = tablet->add_inc_rowset(*rowset);
+            if (publish_status != OLAP_SUCCESS) {
+                LOG(WARNING) << "add visilbe rowset to tablet failed rowset_id:" << rowset->rowset_id()
+                             << "tablet id: " << tablet_info.tablet_id
```

Comment:
add a space or you will see rowset_id followed directly by tablet id string. It is not easy to check log.
the same to the following lines.